tests/framework/microvm.py

```diff
@@ -255,6 +255,18 @@ def kill(self):
                 LOG.exception("Process not found: %d", self.firecracker_pid)
             except FileNotFoundError:
                 LOG.exception("PID file not found")
+
+            # if microvm was spawned then check if it gets killed
+            if self._spawned:
+                # filter ps results for the jailer's unique id
+                rc, stdout, stderr = utils.run_cmd(
+                    f"ps aux | grep {self.jailer.jailer_id}"
+                )
+                assert rc == 0 and stderr == ""
+                # make sure firecracker was killed
+                assert (
```

Comment:
if this is killed in a destructor, will the error propagate to the test run result? Eg marking the test as passed and printing an error out of band isn't very helpful.

Reply:
yes, the error will propagate to the test run result and I think it would be good to know if it passes in few cases and fails in others.

Reply:
This should be fine, we dont do cleanup in destructors (because that caused issues in the past because python do not give strong guarantees about when and how often `__del__` is run). Instead it happens during fixture teardown, and pytest will associate these failures with the correct test.